app/controllers/products_controller.rb

```diff
@@ -1,17 +1,15 @@
 class ProductsController < ApplicationController
   def index
-    @product = Product.all
-  end
-
+    if params[:category] == 'male'
```

Comment:
please refactor it like
if params[:category]
@product = Product.where(category: params[:category])
else
@product = Product.all
*THis is rough code